### DISCUSSION TITLE
Feature Request: add command-line option to disable linenoise

### DISCUSSION THREAD

**Discussion by bon** [duckdb/duckdb#6570] [Ideas]
Duckdb links to the linenoise library for rudimentary in-line editing. Linenoise is presumably preferred to readline for licensing reasons since its functionality falls far short of readline. In the past [piecemeal](https://github.com/duckdb/duckdb/pull/1550) [attempts](https://github.com/duckdb/duckdb/issues/1585) have been made to add to linenoise's functionality in duckdb. Rather than make further additions, a duckdb command-line option to disable linenoise would allow power users to start it using e.g. `rlwrap duckdb -nolinenoise` and get full readline functionality.

**Comment by Mytherin** [comment]
This seems to work for me with the `-batch` command:

```bash
rlwrap duckdb -batch
```

Does that not work for you?

**Reply by bon** [reply]
It appears to work but it suppresses the prompt so it's a little disorienting. Any way to keep the prompt and disable linenoise?

**Reply by Mytherin** [reply]
rlwrap allows you to insert a prompt:

```bash
rlwrap --substitute-prompt="> " duckdb -batch
```

**Reply by bon** [reply]
That works, thanks!

**Comment by Alex-Monahan** [comment]
Do you want me to add this to the docs at all? As a non-native Linux speaker, I can't tell if this is something that lots of folks would like to learn how to do!

**Reply by Mytherin** [reply]
I think adding this to the CLI docs is a great idea.

**Reply by bon** [reply]
Agreed. Using `-batch` in this way was not obvious to me.

**Comment by mskyttner** [comment]
Would this be an [alternative option](https://github.com/jart/bestline) to consider? The tin says: "a fork of linenoise (a popular readline alternative) that fixes its bugs and adds the missing features while reducing binary footprint".

**Reply by Mytherin** [reply]
That's very interesting indeed. We have made some modifications to linenoise (including adding e.g. the syntax highlighting) - but I wouldn't be opposed to checking if we can port that to this library instead.

**Comment by bon** [comment]
The thing is, it's never going to approach the full functionality of readline. There's always going to be a moment where you're typing away merrily and there's that one keybinding missing that screws things up. I'll be sticking with the solution with rlwrap that @Mytherin suggested above.
